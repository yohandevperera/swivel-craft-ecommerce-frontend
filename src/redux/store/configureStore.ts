import {
  applyMiddleware,
  legacy_createStore as createStore,
  compose,
} from "redux";
import thunk from "redux-thunk";
import rootReducer from "../reducers/rootReducer";

/**
 * Description and Usage - This file will include all the redux
 * store configurations
 *
 */

declare global {
  interface Window {
    __REDUX_DEVTOOLS_EXTENSION_COMPOSE__?: typeof compose;
  }
}

const configureStore = () => {
  const middlewares = [thunk];
  const composeEnhancers =
    window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose;
  const enhancers = composeEnhancers(applyMiddleware(...middlewares));
  const store = createStore(rootReducer(), enhancers);

  return store;
};

export default configureStore;
