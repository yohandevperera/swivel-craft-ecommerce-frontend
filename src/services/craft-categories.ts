import { invoke } from "../utils/api-adapter";

/**
 *
 * Usage - This file will be used to communicate to the backend API endpoints
 *
 * Description - This file consists of all the API service call functions for the CraftCategorysCategories resource
 *
 */

/**
 * Description - A global type thats used as
 * CraftCategory add payload and to fetch CraftCategorysCategories as well
 */
export interface CraftCategoryType {
  _id: string;
  name: string;
}

/**
 * Description and Usage - This function will be used
 * to fetch all the CraftCategorysCategories from the backend API endpoint
 */
export async function getAllCraftCategories() {
  return invoke("api/craft-categories/", "get", {});
}

/**
 * Description and Usage - This function will be used
 * to create an CraftCategory using the backend API endpoint
 *
 * @param CraftCategory @typedef Omit<CraftCategoryType, "_id" | "photo">
 */
export async function createCraftCategory(
  craftCategory: Omit<CraftCategoryType, "_id">
) {
  return invoke("api/craft-categories/", "post", craftCategory);
}

/**
 * Description and Usage - This function will be used
 * to edit an CraftCategory using the backend API endpoint
 *
 * @param CraftCategory @typedef Omit<CraftCategoryType, "_id" | "photo">
 * @param id @typedef string
 */
export async function editCraftCategory(
  craftCategory: Omit<CraftCategoryType, "_id" | "photo">,
  id?: string
) {
  return invoke(`api/craft-categories${id}`, "patch", craftCategory);
}

/**
 * Description and Usage - This function will be used
 * to remove an CraftCategory using the backend API endpoint
 *
 * @param id @typedef string
 */
export async function deleteCraftCategory(id: string) {
  return invoke(`api/craft-categories/${id}`, "delete", {});
}

/**
 * Description and Usage - This function will be used
 * to fetch a single CraftCategory for a given id using the backend API endpoint
 *
 * @param id @typedef string
 */
export async function getCraftCategory(id?: string) {
  return invoke(`api/craft-categories/get-craft-category/${id}`, "get", {});
}

export async function searchCraftCategoryByName(name?: string) {
  return invoke(`api/craft-categories/find-by-name/${name}`, "get", {});
}

export async function getCraftCategoriesByName() {
  return invoke(`api/craft-categories/get-category-by-name`, "get", {});
}
