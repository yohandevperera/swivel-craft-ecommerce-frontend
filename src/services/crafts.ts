import { invoke } from "../utils/api-adapter";

/**
 *
 * Usage - This file will be used to communicate to the backend API endpoints
 *
 * Description - This file consists of all the API service call functions for the crafts resource
 *
 */

/**
 * Description - A global type thats used as
 * craft add payload and to fetch crafts as well
 */
export interface CraftType {
  _id: string;
  name: string;
  categoryId: string;
  description: string;
  qty: number;
  price: number;
  photo: any;
}

/**
 * Description and Usage - This function will be used
 * to fetch all the crafts from the backend API endpoint
 */
export async function getAllCrafts() {
  return invoke("api/crafts/", "get", {});
}

/**
 * Description and Usage - This function will be used
 * to create an craft using the backend API endpoint
 *
 * @param craft @typedef Omit<CraftType, "_id" | "photo">
 */
export async function createCraft(craft: Omit<CraftType, "_id" | "photo">) {
  return invoke("api/crafts/", "post", craft);
}

/**
 * Description and Usage - This function will be used
 * to edit an craft using the backend API endpoint
 *
 * @param craft @typedef Omit<CraftType, "_id" | "photo">
 * @param id @typedef string
 */
export async function editCraft(
  craft: Omit<CraftType, "_id" | "photo">,
  id?: string
) {
  return invoke(`api/crafts/${id}`, "patch", craft);
}

/**
 * Description and Usage - This function will be used
 * to remove an craft using the backend API endpoint
 *
 * @param id @typedef string
 */
export async function deleteCraft(id: string) {
  return invoke(`api/crafts/${id}`, "delete", {});
}

/**
 * Description and Usage - This function will be used
 * to fetch a single craft for a given id using the backend API endpoint
 *
 * @param id @typedef string
 */
export async function getCraft(id?: string) {
  return invoke(`api/crafts/${id}`, "get", {});
}
