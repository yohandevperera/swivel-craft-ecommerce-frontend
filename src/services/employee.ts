import { invoke } from "../utils/api-adapter";

/**
 *
 * Usage - This file will be used to communicate to the backend API endpoints
 *
 * Description - This file consists of all the API service call functions
 *
 */

/**
 * Description - A global type thats used as
 * employee add payload and to fetch employees as well
 */
export interface EmployeeType {
  _id: string;
  photo: any;
  firstname: string;
  lastname: string;
  email: string;
  phone: string;
  gender: "M" | "F" | "Male" | "Female" | string;
}

/**
 * Description and Usage - This function will be used
 * to fetch all the employees from the backend API endpoint
 */
export async function getAllEmployees() {
  return invoke("api/employees/", "get", {});
}

/**
 * Description and Usage - This function will be used
 * to create an employee using the backend API endpoint
 *
 * @param employee @typedef Omit<EmployeeType, "_id" | "photo">
 */
export async function createEmployee(
  employee: Omit<EmployeeType, "_id" | "photo">
) {
  return invoke("api/employees/", "post", employee);
}

/**
 * Description and Usage - This function will be used
 * to edit an employee using the backend API endpoint
 *
 * @param employee @typedef Omit<EmployeeType, "_id" | "photo">
 * @param id @typedef string
 */
export async function editEmployee(
  employee: Omit<EmployeeType, "_id" | "photo">,
  id?: string
) {
  return invoke(`api/employees/${id}`, "put", employee);
}

/**
 * Description and Usage - This function will be used
 * to remove an employee using the backend API endpoint
 *
 * @param id @typedef string
 */
export async function deleteEmployee(id: string) {
  return invoke(`api/employees/${id}`, "delete", {});
}

/**
 * Description and Usage - This function will be used
 * to fetch a single employee for a given id using the backend API endpoint
 *
 * @param id @typedef string
 */
export async function getEmployee(id?: string) {
  return invoke(`api/employees/${id}`, "get", {});
}
