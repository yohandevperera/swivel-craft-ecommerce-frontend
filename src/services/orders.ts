import { invoke } from "../utils/api-adapter";

/**
 *
 * Usage - This file will be used to communicate to the backend API endpoints
 *
 * Description - This file consists of all the API service call functions for the Orders resource
 *
 */

/**
 * Description - A global type thats used as
 * Order add payload and to fetch Orders as well
 */
export interface OrderType {
  _id: string;
  orderId: string;
  userId: string;
  totalPrice: number;
  qtyBought: number;
  craftId: string;
}

/**
 * Description and Usage - This function will be used
 * to fetch all the Orders from the backend API endpoint
 */
export async function getAllOrders() {
  return invoke("api/orders/", "get", {});
}

/**
 * Description and Usage - This function will be used
 * to create an Order using the backend API endpoint
 *
 * @param Order @typedef Omit<OrderType, "_id" | "photo">
 */
export async function createOrder(Order: Omit<OrderType, "_id">[]) {
  return invoke("api/orders/", "post", Order);
}

/**
 * Description and Usage - This function will be used
 * to fetch a single Order for a given id using the backend API endpoint
 *
 * @param id @typedef string
 */
export async function getOrder(id?: string) {
  return invoke(`api/orders/${id}`, "get", {});
}
