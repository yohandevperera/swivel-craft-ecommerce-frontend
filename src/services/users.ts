import { invoke } from "../utils/api-adapter";

/**
 *
 * Usage - This file will be used to communicate to the backend API endpoints
 *
 * Description - This file consists of all the API service call functions for the crafts resource
 *
 */

/**
 * Description - A global type thats used as
 * craft add payload and to fetch crafts as well
 */
export interface UserType {
  _id: string;
  firstname: string;
  email: string;
  phone: string;
  password: number;
}

/**
 * Description and Usage - This function will be used
 * to fetch all the crafts from the backend API endpoint
 */
export async function getAllUsers() {
  return invoke("api/users", "get", {});
}

/**
 * Description and Usage - This function will be used
 * to create an craft using the backend API endpoint
 *
 * @param craft @typedef Omit<UserType, "_id" | "photo">
 */
export async function createUser(craft: Omit<UserType, "_id">) {
  return invoke("api/users/", "post", craft);
}

/**
 * Description and Usage - This function will be used
 * to edit an craft using the backend API endpoint
 *
 * @param craft @typedef Omit<UserType, "_id" | "photo">
 * @param id @typedef string
 */
export async function editUser(craft: Omit<UserType, "_id">, id?: string) {
  return invoke(`api/users/${id}`, "patch", craft);
}

/**
 * Description and Usage - This function will be used
 * to remove an craft using the backend API endpoint
 *
 * @param id @typedef string
 */
export async function deleteUser(id: string) {
  return invoke(`api/users/${id}`, "delete", {});
}

/**
 * Description and Usage - This function will be used
 * to fetch a single craft for a given id using the backend API endpoint
 *
 * @param id @typedef string
 */
export async function getUser(id?: string) {
  return invoke(`api/users/${id}`, "get", {});
}

export async function searchUserByFristname(name?: string) {
  return invoke(`api/users/find-by-name/${name}`, "get", {});
}
