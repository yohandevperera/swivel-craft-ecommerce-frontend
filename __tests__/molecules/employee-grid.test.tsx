import "@testing-library/jest-dom";
import React from "react";
import AppRoutes from "../../src/components/organisms/app-routes";
import { EmployeeType } from "../../src/services/employee";
import { BrowserRouter } from "react-router-dom";
import { Provider } from "react-redux";
import configureStore from "../../src/redux/store/configureStore";
import GridItem from "../../src/components/molecules/grid-item";
import renderer from "react-test-renderer";


// Sample data set to check the grid view
const employee: EmployeeType = {
  email: "Darrin_Rippin@gmail.com",
  firstname: "Henri",
  gender: "M",
  lastname: "Rodriguez",
  phone: "+94771277218",
  photo: "https://randomuser.me/api/portraits/men/92.jpg",
  _id: "63ff04ffb8e613034a3ea943",
};

// the store instance created to access the redux store
const store = configureStore();

// test snapshot to validate the data grid view
describe("Employee grid snapshot", () => {
  it("should render the employee gird with the meta test data", () => {
    const domTree = renderer
      .create(
        <>
          <Provider store={store}>
            <BrowserRouter>
              <AppRoutes />
              <GridItem item={employee} handleDelete={() => {}} />
            </BrowserRouter>
          </Provider>
        </>
      )
      .toJSON();
    expect(domTree).toMatchSnapshot();
  });
});
