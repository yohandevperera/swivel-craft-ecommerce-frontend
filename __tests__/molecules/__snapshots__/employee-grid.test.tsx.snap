// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Employee grid snapshot should render the employee gird with the meta test data 1`] = `
[
  <div
    className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-1bwk6yd-MuiPaper-root-MuiCard-root"
  >
    <div
      className="css-7s17j4"
    >
      Employee Manager
    </div>
  </div>,
  <div
    className="MuiContainer-root MuiContainer-maxWidthLg css-1oqqzyl-MuiContainer-root"
  >
    <div
      className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-oonc48-MuiPaper-root-MuiCard-root"
    >
      <div
        aria-owns={null}
        className="MuiAutocomplete-root css-4xbbh5-MuiAutocomplete-root"
        onClick={[Function]}
        onKeyDown={[Function]}
        onMouseDown={[Function]}
        style={
          {
            "float": "left",
            "marginBottom": 10,
            "width": "30%",
          }
        }
      >
        <div
          className="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
        >
          <label
            className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeSmall MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeSmall MuiInputLabel-outlined css-1pysi21-MuiFormLabel-root-MuiInputLabel-root"
            data-shrink={false}
            htmlFor="employee-search"
            id="employee-search-label"
          >
            Search employee by firstname
          </label>
          <div
            className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-sizeSmall MuiAutocomplete-inputRoot css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
            onClick={[Function]}
          >
            <input
              aria-activedescendant={null}
              aria-autocomplete="list"
              aria-expanded={false}
              aria-invalid={false}
              autoCapitalize="none"
              autoComplete="off"
              autoFocus={false}
              className="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputTypeSearch MuiInputBase-inputSizeSmall MuiAutocomplete-input MuiAutocomplete-inputFocused css-1o6kl88-MuiInputBase-input-MuiOutlinedInput-input"
              id="employee-search"
              onAnimationStart={[Function]}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              onMouseDown={[Function]}
              readOnly={false}
              required={false}
              role="combobox"
              spellCheck="false"
              type="search"
              value=""
            />
            <fieldset
              aria-hidden={true}
              className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
            >
              <legend
                className="css-yjsfm1"
              >
                <span>
                  Search employee by firstname
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
      </div>
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium css-sghohy-MuiButtonBase-root-MuiButton-root"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        style={
          {
            "marginLeft": 10,
          }
        }
        tabIndex={0}
        type="button"
      >
        Search
      </button>
      <div
        onClick={[Function]}
      >
        <label
          aria-label="upload picture"
          className="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorPrimary MuiIconButton-sizeMedium css-1kuq5xv-MuiButtonBase-root-MuiIconButton-root"
          onBlur={[Function]}
          onContextMenu={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          role="button"
          tabIndex={0}
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="RefreshIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M17.65 6.35C16.2 4.9 14.21 4 12 4c-4.42 0-7.99 3.58-7.99 8s3.57 8 7.99 8c3.73 0 6.84-2.55 7.73-6h-2.08c-.82 2.33-3.04 4-5.65 4-3.31 0-6-2.69-6-6s2.69-6 6-6c1.66 0 3.14.69 4.22 1.78L13 11h7V4l-2.35 2.35z"
            />
          </svg>
        </label>
      </div>
      <div
        onClick={[Function]}
        style={{}}
      >
        <button
          className="MuiButtonBase-root Mui-disabled MuiIconButton-root Mui-disabled MuiIconButton-colorPrimary MuiIconButton-sizeMedium css-1kuq5xv-MuiButtonBase-root-MuiIconButton-root"
          disabled={true}
          onBlur={[Function]}
          onContextMenu={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={-1}
          type="button"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="SortByAlphaIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M14.94 4.66h-4.72l2.36-2.36zm-4.69 14.71h4.66l-2.33 2.33zM6.1 6.27 1.6 17.73h1.84l.92-2.45h5.11l.92 2.45h1.84L7.74 6.27H6.1zm-1.13 7.37 1.94-5.18 1.94 5.18H4.97zm10.76 2.5h6.12v1.59h-8.53v-1.29l5.92-8.56h-5.88v-1.6h8.3v1.26l-5.93 8.6z"
            />
          </svg>
        </button>
      </div>
      <div
        onClick={[Function]}
        style={
          {
            "float": "right",
          }
        }
      >
        <button
          className="MuiButtonBase-root Mui-disabled MuiIconButton-root Mui-disabled MuiIconButton-colorPrimary MuiIconButton-sizeMedium css-1kuq5xv-MuiButtonBase-root-MuiIconButton-root"
          disabled={true}
          onBlur={[Function]}
          onContextMenu={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={-1}
          type="button"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="TocIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M3 9h14V7H3v2zm0 4h14v-2H3v2zm0 4h14v-2H3v2zm16 0h2v-2h-2v2zm0-10v2h2V7h-2zm0 6h2v-2h-2v2z"
            />
          </svg>
        </button>
      </div>
      <a
        className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-1e6y48t-MuiButtonBase-root-MuiButton-root"
        href="/employee-add"
        onBlur={[Function]}
        onClick={[Function]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        style={
          {
            "float": "right",
          }
        }
        tabIndex={0}
      >
        Add Employee
      </a>
      <span
        className="MuiCircularProgress-root MuiCircularProgress-indeterminate MuiCircularProgress-colorPrimary css-18lrjg1-MuiCircularProgress-root"
        role="progressbar"
        style={
          {
            "height": 40,
            "marginLeft": "50%",
            "marginTop": "10%",
            "width": 40,
          }
        }
      >
        <svg
          className="MuiCircularProgress-svg css-1idz92c-MuiCircularProgress-svg"
          viewBox="22 22 44 44"
        >
          <circle
            className="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate css-176wh8e-MuiCircularProgress-circle"
            cx={44}
            cy={44}
            fill="none"
            r={20.2}
            strokeWidth={3.6}
            style={{}}
          />
        </svg>
      </span>
    </div>
  </div>,
  <div
    className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 css-cz2w44-MuiPaper-root"
  >
    <div
      className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 css-mhc70k-MuiGrid-root"
    >
      <div
        className="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
      >
        <div
          className="MuiBox-root css-5u3pr2"
        >
          <img
            alt="complex"
            className="css-y9s1zk"
            src="https://randomuser.me/api/portraits/men/92.jpg"
          />
        </div>
      </div>
      <div
        className="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-true css-114x5zz-MuiGrid-root"
      >
        <div
          className="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-spacing-xs-2 MuiGrid-direction-xs-column MuiGrid-grid-xs-true css-hfwr93-MuiGrid-root"
        >
          <div
            className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-true css-1vd824g-MuiGrid-root"
          >
            <div
              className="MuiTypography-root MuiTypography-subtitle1 MuiTypography-gutterBottom css-1wniyei-MuiTypography-root"
            >
              Henri   Rodriguez
            </div>
            <p
              className="MuiTypography-root MuiTypography-body2 MuiTypography-gutterBottom css-ejgmvm-MuiTypography-root"
            >
              Darrin_Rippin@gmail.com
            </p>
            <p
              className="MuiTypography-root MuiTypography-body2 css-r40f8v-MuiTypography-root"
            >
              +94771277218
            </p>
            <p
              className="MuiTypography-root MuiTypography-body2 css-r40f8v-MuiTypography-root"
            >
              Male
            </p>
          </div>
          <div
            className="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
          >
            <div
              className="MuiStack-root css-vr01dl-MuiStack-root"
            >
              <a
                className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-1e6y48t-MuiButtonBase-root-MuiButton-root"
                href="/employee-edit/63ff04ffb8e613034a3ea943"
                onBlur={[Function]}
                onClick={[Function]}
                onContextMenu={[Function]}
                onDragLeave={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                tabIndex={0}
              >
                <svg
                  aria-hidden={true}
                  className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                  data-testid="EditIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M3 17.25V21h3.75L17.81 9.94l-3.75-3.75L3 17.25zM20.71 7.04c.39-.39.39-1.02 0-1.41l-2.34-2.34a.9959.9959 0 0 0-1.41 0l-1.83 1.83 3.75 3.75 1.83-1.83z"
                  />
                </svg>
              </a>
              <div
                onClick={[Function]}
                style={{}}
              >
                <button
                  className="MuiButtonBase-root Mui-disabled MuiIconButton-root Mui-disabled MuiIconButton-colorPrimary MuiIconButton-sizeMedium css-1kuq5xv-MuiButtonBase-root-MuiIconButton-root"
                  disabled={true}
                  onBlur={[Function]}
                  onContextMenu={[Function]}
                  onDragLeave={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  onKeyUp={[Function]}
                  onMouseDown={[Function]}
                  onMouseLeave={[Function]}
                  onMouseUp={[Function]}
                  onTouchEnd={[Function]}
                  onTouchMove={[Function]}
                  onTouchStart={[Function]}
                  tabIndex={-1}
                  type="button"
                >
                  <svg
                    aria-hidden={true}
                    className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                    data-testid="DeleteIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM19 4h-3.5l-1-1h-5l-1 1H5v2h14V4z"
                    />
                  </svg>
                </button>
              </div>
            </div>
            <div />
          </div>
        </div>
      </div>
    </div>
  </div>,
]
`;
